.github/workflows/ci-build.yml

```diff
@@ -0,0 +1,17 @@
+name: CI
+
+on:
+   pull_request:
+    branches: [ main ]
+   
+jobs:
+  Code-Quality-Checks:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.8'
+    - name: Run linter
```

Comment:
I've also used `pydocstyle` during development which enforces PEP357 (docstring validations). AFAIK `pylint` does not support that one so we need to run both tools in the linter, one for coding style and one for docstrings.

Also I'm okay with us switching to pylint but then you need to remove the `pycodestyle` dependency since it does the same thing essentially.